app/src/main/java/com/example/vmedvediev/redditapp/PostsRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,103 @@
+package com.example.vmedvediev.redditapp
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.ProgressBar
+import android.widget.TextView
+import com.example.vmedvediev.redditapp.model.Post
+import com.nostra13.universalimageloader.cache.memory.impl.WeakMemoryCache
+import com.nostra13.universalimageloader.core.DisplayImageOptions
+import com.nostra13.universalimageloader.core.ImageLoader
+import com.nostra13.universalimageloader.core.ImageLoaderConfiguration
+import com.nostra13.universalimageloader.core.assist.FailReason
+import com.nostra13.universalimageloader.core.assist.ImageScaleType
+import com.nostra13.universalimageloader.core.display.FadeInBitmapDisplayer
+import com.nostra13.universalimageloader.core.listener.ImageLoadingListener
+import org.jetbrains.anko.find
+
+class PostsRecyclerViewAdapter(val context: Context, val postsList: ArrayList<Post>) : RecyclerView.Adapter<PostsRecyclerViewAdapter.ViewHolder>() {
+
+    companion object {
+        private const val TAG = "PostsRecyclerViewAdapter"
+    }
+
+    init {
+        setupImageLoader()
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(context).inflate(R.layout.item_post, parent, false)
+        return ViewHolder(view)
+    }
+
+    override fun getItemCount(): Int = postsList.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.title.text = postsList[position].title
+        holder.author.text = postsList[position].author
+        holder.dateUpdated.text = postsList[position].dateUpdated
+
+        val imageLoader = ImageLoader.getInstance()
```

Comment:
Extract all work with image loader to a separate function, e.g., `showImage(...)` (function name - your desicion)